docs/source/format/Metadata.rst

```diff
@@ -390,7 +390,25 @@ Time supports the same unit resolutions: second, millisecond, microsecond, and
 nanosecond. We represent time as the smallest integer accommodating the
 indicated unit. For second and millisecond: 32-bit, for the others 64-bit.
 
-Dictionary encoding
--------------------
+Extension Types
+---------------
+
+User-defined "extension" types can be defined in a
+language-independent way by setting special ``KeyValue`` pairs in the
+``custom_metadata`` field of the ``Field`` metadata structure. These
+special extension keys are:
+
+* ``'ARROW:extension_name'`` for the string name identifying the
```

Comment:
can we formalize that "ARROW:" is a reserved key prefix for metadata.

Reply:
Yes, I'll do that